File: Scripts/newTask/inheritance-single/c++/inheritance-single.cpp
class Animal
{
  // ...
};

class Dog: public Animal
{
  // ...
};

class Lab: public Dog
{
  // ...
};

class Collie: public Dog
{
  // ...
};

class Cat: public Animal
{
  // ...
};


int main() {

  for ( int i = 0; i < 10000000; ++i ) {
    Cat kitty;
    Dog doggy;
    Lab laboratory;
    Collie collie;
  }

  return 0;
}

File: Scripts/newTask/inheritance-single/php/inheritance-single.php
<?php

class Animal {
   // functions go here...
   function speak() {
    echo "Animal speaking \n";
    }

    function walk() {
        echo "Animal walking \n";
    }
}

class Dog extends Animal {
   // functions go here...
    function speak() {
        echo "Dog barking \n";
    }
}

class Cat extends Animal {
   // functions go here...
    function speak() {
        echo "Cat meawing \n";
    }
}

class Lab extends Dog {
   // functions go here...
    function speak() {
        echo "Lab barking \n";
    }
}

class Collie extends Dog {
   // functions go here...
    function speak() {
        echo "Collie barking \n";
    }
}

for ( $i = 0; $i < 1000000; $i++) {
	$myObj = new Dog();
    $myObj->speak();
    $myObj->walk();
}

?>